CommandScreen.c

```diff
@@ -12,45 +12,124 @@ in the source distribution for its full text.
+   for (size_t i = 0; i < total; i++, line_offset++) {
+      assert(line_offset >= 0 && (size_t)line_offset <= total);
+      char c = line[line_offset] = p[i];
+      if (c == ' ') {
+         last_spc_offset = line_offset;
       }
 
       if (line_offset == COLS) {
-         len = (last_spc == -1) ? line_offset : last_spc;
-         line[len] = '\0';
+         line_size = (last_spc_offset == -1) ? line_offset : last_spc_offset;
+         line[line_size] = '\0';
          InfoScreen_addLine(this, line);
 
-         line_offset -= len;
-         last_spc = -1;
-         memcpy(line, p - line_offset, line_offset + 1);
+         line_offset -= line_size;
+         last_spc_offset = -1;
+         memcpy(line, p + i - line_offset, line_offset + 1);
+      }
+   }
+
+   return line_offset;
+}
+
+#ifdef HAVE_LIBNCURSESW
+
+static int CommandScreen_scanWide(InfoScreen* this, const char* p, size_t total, char* line) {
+   mbstate_t state;
+   memset(&state, 0, sizeof(state));
+   int line_cols = 0, line_offset = 0, line_size = 0, width = 1;
+   int last_spc_cols = -1, last_spc_offset = -1;
+   for (size_t i = 0, bytes = 1; i < total; bytes = 1, width = 1) {
+      assert(line_offset >= 0 && (size_t)line_offset <= total);
+      unsigned char c = (unsigned char)p[i];
+      if (c < 0x80) { // skip mbrtowc for ASCII characters
+         line[line_offset] = c;
+         if (c == ' ') {
+            last_spc_offset = line_offset;
+            last_spc_cols = line_cols;
+         }
+      } else {
+         wchar_t wc;
+         bytes = mbrtowc(&wc, p + i, total - i, &state);
+         if (bytes != (size_t)-1 && bytes != (size_t)-2) {
+            width = wcwidth(wc);
+            width = MAXIMUM(width, 1);
+         } else {
+            bytes = 1;
+         }
+         memcpy(line + line_offset, p + i, bytes);
+      }
+
+      i += bytes;
+      line_offset += bytes;
+      line_cols += width;
+      if (line_cols < COLS) {
+         continue;
+      }
+
+      if (last_spc_offset >= 0) {  // wrap by last space
+         line_size = last_spc_offset;
+         line_cols -= last_spc_cols;
+         last_spc_offset = last_spc_cols = -1;
+      } else if (line_cols > COLS) {  // wrap current wide char to next line
+         line_size = line_offset - (int) bytes;
+         line_cols = width;
+      } else { // wrap by current character
+         line_size = line_offset;
+         line_cols = 0;
+      }
+
+      line[line_size] = '\0';
+      InfoScreen_addLine(this, line);
+      line_offset -= line_size;
+      if (line_offset > 0) {
+         memcpy(line, p + i - line_offset, line_offset + 1);
       }
    }
 
+   return line_offset;
+}
+
+#endif // HAVE_LIBNCURSESW
+
+static void CommandScreen_scan(InfoScreen* this) {
+   Panel* panel = this->display;
+   int idx = Panel_getSelectedIndex(panel);
+   Panel_prune(panel);
+
+   const char* p = Process_getCommand(this->process);
+   assert(p != NULL);
+   size_t total = strlen(p);
+   char line[total + 1];
```

Comment:
Avoid variable length arrays.

Reply:
Preferably similar to the `String_split` function or with a callback as suggested.

Reply:
> If a first version only covers the cases used on CommandScreen.c that's totally fine. @Explorer09 can take up the work from there if you like.

Yes, that would be great, last time I used C was back when I created the initial `CommandScreen` implmentation :-)

Reply:
I made `String_lineBreakWidth()` in https://github.com/htop-dev/htop/pull/1648/commits/2fbe8f2a6e012130f6a03a7d63974608fcb2edd9 that should be able to work with your use case.